File: problem_22/names_scores_test.rb
require 'minitest/test'
require 'minitest/autorun'

class NamesScoresTest < Minitest::Test
  def names_scores(names)

  end

  def alphabetical_value(name)
    name.chars.map { |letter| letter.ord - 64 }.sum
  end

  #----------------- Tests -----------------#

  def test_names_scores
    assert_equal 504, names_scores(NAMES)
  end

  def test_alphabetical_value
    assert_equal 53, alphabetical_value('COLIN')
    assert_equal 57, alphabetical_value('MARY')
    assert_equal 77, alphabetical_value('PATRICIA')
  end
end

File: problem_22/names-scores.test.js
//----------------- Constants -----------------//

const names = require("./names");

//----------------- Functions -----------------//

function namesScores(nameList) {
  let sum = 0;

  nameList.sort().forEach((name, index) => {
    sum += alphabeticalValue(name) * (index + 1);
  });

  return sum;
}

function alphabeticalValue(name) {
  let value = 0;

  for (i = 0; i < name.length; i++) {
    value += name.charCodeAt(i) - 64;
  }

  return value;
}

//----------------- Tests -----------------//

test("namesScores", () => {
  expect(namesScores(["COLIN", "MARY", "PATRICIA"])).toBe(398);
  expect(namesScores(names)).toBe(871198282);
});

test("alphabeticalValue", () => {
  expect(alphabeticalValue("COLIN")).toBe(53);
  expect(alphabeticalValue("MARY")).toBe(57);
  expect(alphabeticalValue("PATRICIA")).toBe(77);
});
